### PR TITLE
chore: notarize Mac applications

### LINKED ISSUE #51: [Script Recorder] Notarise Electron App
On macOS, we need to notarise our application to prevent Gatekeeper from throwing an error when it's run that requires a user to process a security exception. This requires admin access which a lot of users won't have, is a frustrating step for users who do have permissions and is reputation ally damaging.

https://kilianvalkhof.com/2019/electron/notarizing-your-electron-application/

### PR DESCRIPTION
**Pull request by kyungeunni**
closes #51 

## Summary
Notarize Mac applications so that the users can use the app without worrying about malicious code.

This PR introduces [electron-notarize](https://github.com/electron/electron-notarize) module to notarize the app and it is used when the Electron builder's `afterSign` hook is called. It is a recommended by electron-builder maintainers.

## How to validate the change
After the artefacts are released, download them, and run the following commands to verify the app has been notarized.

(Artifacts generated from this branch can be found [here](https://github.com/elastic/synthetics-recorder/releases/tag/untagged-053cedd0e62b7b3b9d3f))

- Validate the app is notarized(should see _source=**Notarized Developer ID**_):
```
❯ spctl -a -t exec -vv $PATH_TO_APP/Elastic Synthetics Recorder.app
$PATH_TO_APP//Elastic Synthetics Recorder.app: accepted
source=Notarized Developer ID
origin=Developer ID Application: Elasticsearch, Inc (2BT3HPN62Z)
```

- Check the ticket is stapled(should see **_The validate action worked!_**):
```
❯ xcrun stapler validate $PATH_TO_APP/Elastic Synthetics Recorder.app
Processing: $PATH_TO_APP/Elastic Synthetics Recorder.app
The validate action worked!
```
- When opening the app, the warning message _Elastic Synthetics Recorder" can't be opened because Apple cannot check it for malicious software_ shouldn't be shown, instead, the message could be something like this:
image

#### Checklist
I've check above points for all the artifacts below
- [x] arm64.dmg (after installing)
- [x] arm64.zip
- [x] x64.dmg (after installing)
- [x] x64.zip

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -306,10 +306,14 @@ def runReleaseWithKeychain() {
   }
   def token = getVaultSecret.getVaultToken(vault.vault_addr, vault.vault_role_id, vault.vault_secret_id)
   def apple = getVaultSecret.getVaultSecretObject(vault.vault_addr, 'secret/release/apple', token).data
-  if (!apple.containsKey('keychain-password')) {
-    error("runReleaseWithKeychain: apple secret could not be accessed correctly")
+  if (!apple.containsKey('keychain-password') || !apple.containsKey('username') || !apple.containsKey('app-specific-password')) {
```
Comment (v1v) [inline .ci/Jenkinsfile:309]: IIUC, those fields are already available in the existing Vault secret defined in the `L308`, is that correct?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "synthetics-recorder",
-  "version": "0.0.1-beta.6",
+  "version": "0.0.1-notarization.3",
```
Comment (v1v) [inline package.json:3]: is this the expected version?

**Review by v1v — APPROVED** [review]
LGTM, just a couple of minor questions

File: .ci/Jenkinsfile (hunk 1 above)
Comment (kyungeunni) [reply .ci/Jenkinsfile:309]: Yes that's correct.

File: package.json (hunk 2 above)
Comment (kyungeunni) [reply package.json:3]: Good catch, I bumped the version during the testing, will revert it 👍

**Review by v1v — APPROVED** [review]

**Pull request merged**